### DISCUSSION TITLE
No root span on some traces using Opentelemetry

### DISCUSSION THREAD

**Discussion by bradleyberwick** [vercel/next.js#61505] [Help]
### Summary

I'm using Opentelemetry to send traces and metrics from my Next.js app to Grafana. For all API requests and *.json requests, the traces do no have a root span i.e. there's no span with 'parentSpanID = 0000000000000000'. This is an issue as I can't view the traces correctly in Grafana. My Opentelemetry instrumentation code is below.

**Is there an issue with my instrumentation or is this an issue with Next.js?**

I'm pretty new to Next.js so if there's any other helpful information which I've not included here, please let me know.

### Additional information

```js
const { NodeSDK } = require('@opentelemetry/sdk-node');
const { PeriodicExportingMetricReader, ConsoleMetricExporter } = require('@opentelemetry/sdk-metrics');
const { OTLPTraceExporter } = require('@opentelemetry/exporter-trace-otlp-grpc');
const { OTLPMetricExporter } = require('@opentelemetry/exporter-metrics-otlp-grpc');
const { getNodeAutoInstrumentations } = require('@opentelemetry/auto-instrumentations-node');

const sdk = new NodeSDK({
// Autoinstrumentation (run in App Service) - tested this and like the OTEL frontend app, it doesn't provide correct request names in traces and there's also no http.route label for http metrics
  traceExporter: new OTLPTraceExporter(), // new ConsoleSpanExporter()
  metricReader: new PeriodicExportingMetricReader({
    exporter: new OTLPMetricExporter(), // new ConsoleMetricExporter()
  }),
  instrumentations: [
    getNodeAutoInstrumentations({
      '@opentelemetry/instrumentation-fs': {
        enabled: false,
      },
      '@opentelemetry/instrumentation-http': {
          // This gives your request spans a more meaningful name than `HTTP GET`
          requestHook: (span, request) => {
            span.updateName( `${request.method} ${request.url || request.path}` );
            // The below is only working on spans with kind=client, not kind=server
            span.setAttribute('http.route', request.path);
          },
       }
    })
  ]
});

sdk.start();
```

**Comment by StephenGoodall** [comment]
@bradleyberwick did you manage to find a workaround for this issue? I think I'm coming across the same one, I get lots of spans, and they do show up as being connected to each other - but none of them are marked as root spans, which makes it hard to make sense of them.

I've tried with the vercel/otel option and the opentelemetry version (like your example) and I think it behaves the same either way.

PS. If you don't mind, I might have to borrow that requestHook! 😃
